src/WebJobs.Extensions.CosmosDB/Config/CosmosDBConfiguration.cs

```diff
@@ -58,34 +60,30 @@ public void Initialize(ExtensionConfigContext context)
 
             INameResolver nameResolver = context.Config.GetService<INameResolver>();
 
-            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
-
             // Use this if there is no other connection string set.
             _defaultConnectionString = nameResolver.Resolve(AzureWebJobsCosmosDBConnectionStringName);
 
-            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
-
-            IBindingProvider outputProvider = factory.BindToCollector<CosmosDBAttribute, OpenType>(typeof(CosmosDBCollectorBuilder<>), this);
-
-            IBindingProvider outputProviderJObject = factory.BindToCollector<CosmosDBAttribute, JObject>(typeof(CosmosDBCollectorBuilder<>), this);
-
-            IBindingProvider clientProvider = factory.BindToInput<CosmosDBAttribute, DocumentClient>(new CosmosDBClientBuilder(this));
+            // Apply ValidateConnection to all on this rule. 
+            var rule = context.AddBindingRule<CosmosDBAttribute>();
+            rule.AddValidator(ValidateConnection);
+            rule.BindToCollector<OpenType>(typeof(CosmosDBCollectorBuilder<>), this);
+            rule.BindToCollector<JObject>(typeof(CosmosDBCollectorBuilder<>), this);
```

Comment:
We shouldn't need this `JObject` one anymore, right? I added this as the workaround for the 'default type' issue in javascript.

Reply:
Correct. I can remove it. I also added a test in SDK to verify this. 

---
In reply to: [151225522](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/331#discussion_r151225522) [](ancestors = 151225522)